File: Kap13/JavaScript/Listing39_47/mygraphs/Kreis.mjs
import ZweiDElm from "./ZweiDElm.mjs";

export default class Kreis extends ZweiDElm {
    #radius;
     
    constructor(initRandfarbe, initFuellfarbe, radius) {
      super(initRandfarbe, initFuellfarbe);
      this.#radius = radius;
      this.berechneUmfang();
      this.berechneFlaeche();
      console.log("Kreis erzeugt.");
    }
    
    berechneUmfang() {
        this.umfang = 2*this.#radius * Math.PI;
    }

    berechneFlaeche() {
        this.flaeche = this.#radius * this.#radius * Math.PI;
    }

    setRadius(radius) {
        this.#radius = radius;
    }
    
    getRadius() {
      return this.#radius;
    }

    printInfo() {
      super.printInfo();
      console.log("Radius: " + this.#radius);
    }
}

File: Kap13/Java/Listing24_26/mygraphs/Kreis.java
package Kap13.Java.Listing24_26.mygraphs;

public class Kreis extends ZweiDElm {
    protected double radius;
     
    public Kreis(int initRandfarbe, int initFuellfarbe, double radius) {
      super(initRandfarbe, initFuellfarbe);
      this.radius = radius;
      berechneUmfang();
      berechneFlaeche();
      System.out.println("Kreis erzeugt.");
    }

    private void berechneUmfang() {
      umfang = 2*radius * Math.PI;
    }

    private void berechneFlaeche() {
      flaeche = radius * radius * Math.PI;
    }

    public void setRadius(double radius) {
        this.radius = radius;
    }
    
    public double getRadius() {
      return radius;
    }

      public void printInfo() {
      super.printInfo();
      System.out.println("Radius: " + radius);
    }
  }	
  